packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -438,12 +436,13 @@ GridRow.propTypes = {
   index: PropTypes.number.isRequired,
   isLastVisible: PropTypes.bool,
   lastColumnToRender: PropTypes.number.isRequired,
-  renderedColumns: PropTypes.arrayOf(PropTypes.object).isRequired,
   row: PropTypes.object.isRequired,
   rowHeight: PropTypes.oneOfType([PropTypes.oneOf(['auto']), PropTypes.number]).isRequired,
   rowId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
   selected: PropTypes.bool.isRequired,
   visibleColumns: PropTypes.arrayOf(PropTypes.object).isRequired,
 } as any;
 
-export { GridRow };
+const MemoizedRow = React.memo(GridRow);
+
+export { MemoizedRow as GridRow };
```

Comment:
Turned out memoizing `GridRow` is problematic.
For example: https://deploy-preview-5978--material-ui-x.netlify.app/x/react-data-grid/row-grouping/


https://user-images.githubusercontent.com/13808724/187712636-f90cbed6-d8ab-48d7-8a1a-4b62c7374c51.mov

Reply:
cc @m4theushw